torch/tensor.py

```diff
@@ -178,6 +178,8 @@ def trim(str):
       Returned Tensor uses the same data tensor as the original one.
       In-place modifications on either of them will be seen, and may trigger
       errors in correctness checks.
+      In-place size / stride / storage changes (such as `resize_` / `resize_as_` / `set_` / `transpose_`)
+      on the returned tensor are not allowed, to ensure consistency with the original tensor.
```

Comment:
I think we can improve the documentation here.

1. The docs say "Tensor uses the same data tensor" as the original one, but that's not really an accurate in the new world. We should say that they share storage.
2. WHY are in-place metadata changes not allowed? You say, "to ensure consistency with the original tensor". But why does that matter? I think the real motivation here, is that we MUST make a BC-breaking change here; and rather than silently break BC, we have opted to make it fail loudly in this case.